elasticsearch/src/http/transport.rs

```diff
@@ -595,16 +701,134 @@ impl CloudConnectionPool {
+    inner: Arc<RwLock<MultiNodeConnectionPoolInner>>,
+    reseed_frequency: Option<Duration>,
+    load_balancing_strategy: LoadBalancing,
+    reseeding: Arc<AtomicBool>,
+}
+
+#[derive(Debug, Clone)]
+pub struct MultiNodeConnectionPoolInner {
+    last_update: Option<Instant>,
+    connections: Vec<Connection>,
+}
+
+impl<LoadBalancing> ConnectionPool for MultiNodeConnectionPool<LoadBalancing>
+where
+    LoadBalancing: LoadBalancingStrategy + Clone,
+{
+    fn next(&self) -> Connection {
+        let inner = self.inner.read().expect("lock poisoned");
+        self.load_balancing_strategy
+            .try_next(&inner.connections)
+            .unwrap()
+    }
+
+    fn reseedable(&self) -> bool {
+        let inner = self.inner.read().expect("lock poisoned");
+        let reseed_frequency = match self.reseed_frequency {
+            Some(wait) => wait,
+            None => return false,
+        };
+        let last_update_is_stale = inner
+            .last_update
+            .as_ref()
+            .map(|last_update| last_update.elapsed() > reseed_frequency);
+        let reseedable = last_update_is_stale.unwrap_or(true);
+
+        return if !reseedable {
+            false
+        } else {
+            // Check if refreshing is false if so, sets to true atomically and returns old value (false) meaning refreshable is true
+            // If refreshing is set to true, do nothing and return true, meaning refreshable is false
+            !self
+                .reseeding
+                .compare_and_swap(false, true, Ordering::Relaxed)
+        };
+    }
+
+    fn reseed(&self, mut connection: Vec<Connection>) {
+        let mut inner = self.inner.write().expect("lock poisoned");
+        inner.last_update = Some(Instant::now());
+        inner.connections.clear();
+        inner.connections.append(&mut connection);
+        self.reseeding.store(false, Ordering::Relaxed);
+    }
+}
+
+impl MultiNodeConnectionPool<RoundRobin> {
+    /** Use a round-robin strategy for balancing traffic over the given set of nodes. */
+    pub fn round_robin(urls: Vec<Url>, reseed_frequency: Option<Duration>) -> Self {
+        let connections = urls.into_iter().map(Connection::new).collect();
+
+        let inner: Arc<RwLock<MultiNodeConnectionPoolInner>> =
+            Arc::new(RwLock::new(MultiNodeConnectionPoolInner {
+                last_update: None,
+                connections,
+            }));
+        let reseeding = Arc::new(AtomicBool::new(false));
+
+        let load_balancing_strategy = RoundRobin::default();
+        Self {
+            inner,
+            load_balancing_strategy,
+            reseed_frequency,
+            reseeding,
+        }
+    }
+}
+
+/** The strategy selects an address from a given collection. */
+pub trait LoadBalancingStrategy: Send + Sync + Debug {
+    /** Try get the next connection. */
+    fn try_next<'a>(&self, connections: &'a [Connection]) -> Result<Connection, Error>;
```

Comment:
```suggestion
    fn try_next(&self, connections: &[Connection]) -> Result<Connection, Error>;
```

The lifetimes can be elided